gdxpds/gdx.py

```diff
@@ -267,6 +267,74 @@ def read(self,filename):
                 symbol.load()
         return
 
+    def read_single_symbol(self,filename,target_symbol_name):
+        """
+        Optimized read method that only loads metadata for a specific symbol.
+
+        Parameters
+        ----------
+        filename : pathlib.Path or str
+            Path to the GDX file to read
+        target_symbol_name : str
+            Name of the specific symbol to read
+
+        Returns
+        -------
+        GdxSymbol
+            The requested symbol (not yet loaded with data)
+
+        Raises
+        ------
+        Error
+            If not self.empty or if symbol not found
+        """
+        if not self.empty:
+            raise Error("GdxFile.read_single_symbol can only be used if the GdxFile is .empty")
+
+        # open the file
+        rc = gdxcc.gdxOpenRead(self.H, str(filename))
+        if not rc[0]:
+            raise GdxError(self.H, f"Could not open {filename!r}")
+        self._filename = filename
+
+        # read in meta-data for the file
+        ret, self._version, self._producer = gdxcc.gdxFileVersion(self.H)
+        if ret != 1:
+            raise GdxError(self.H, "Could not get file version")
+        ret, symbol_count, element_count = gdxcc.gdxSystemInfo(self.H)
+        logger.debug(f"Opening '{filename}' with {symbol_count} symbols and "
+                    f"{element_count} elements, searching for '{target_symbol_name}' with optimized read.")
+
+        # read universal set
+        ret, name, dims, data_type = gdxcc.gdxSymbolInfo(self.H, 0)
+        if ret != 1:
+            raise GdxError(self.H, "Could not get symbol info for the universal set")
+        self.universal_set = GdxSymbol(name, data_type, dims=dims, file=self, index=0)
+
+        # search for target symbol without creating objects for others
+        target_symbol = None
+        for i in range(symbol_count):
+            index = i + 1
+            ret, name, dims, data_type = gdxcc.gdxSymbolInfo(self.H, index)
+            if ret != 1:
+                raise GdxError(self.H, f"Could not get symbol info for symbol {index}")
+
+            if name == target_symbol_name:
+                # found our target - create the symbol object with full metadata
+                try:
+                    target_symbol = GdxSymbol(name, data_type, dims=dims, file=self, index=index)
+                    self.append(target_symbol)
```

Comment:
Is skipping this for symbols whose data you don't want to load costly enough in terms of time to warrant this separate function?